Lib/test/test_shutil.py

```diff
@@ -1403,6 +1403,16 @@ def test_copyfile_same_file(self):
         # But Error should work too, to stay backward compatible.
         self.assertRaises(Error, shutil.copyfile, src_file, src_file)
 
+    def test_copy_w_different_length(self):
+        # copy and copy2 both accept an alternate buffer `length`
+        for fn in (shutil.copy, shutil.copy2):
+            src_dir = self.mkdtemp()
+            dst_dir = self.mkdtemp()
+            src = os.path.join(src_dir, 'foo')
+            write_file(src, 'foo' * 10 ** 6)
+            # buffer with 20 Kib
+            fn(src, dst_dir, length=20 * 1024)
```

Comment:
I suggest to use a smaller file: write('x' * 100) and then length=20.

By the way, I suggest to use tempfile.NamedTemporaryFile rather than temporary files:

```
            with tempfile.NamedTemporaryFile() as src:
                with tempfile.NamedTemporaryFile() as dst:
                    write_file(src.name, b'x' * 100, binary=True)
                    fn(src.name, dst.name, length=20)
```

Reply:
Hehe thanks @vstinner! You did the work for me 👍